### PR TITLE
refactor: replace non-empty QString constructors with QStringLiteral()

### PR DESCRIPTION
**Pull request by Integral-Tech**
- Qt provides a macro [QStringLiteral()](https://doc.qt.io/qt-6/qstring.html#QStringLiteral), which makes constructing QString objects from string literals more efficient.

### CODE REVIEW

File: src/entities/notefolder.cpp [hunk 1]
```diff
@@ -530,12 +530,12 @@ bool NoteFolder::isPathNoteFolder(const QString &path) {
 
 void NoteFolder::setSettingsValue(const QString &key, const QVariant &value) {
     SettingsService settings;
-    settings.setValue(QString("NoteFolder-%1/%2").arg(QString::number(id), key), value);
```
Comment (pbek) [inline src/entities/notefolder.cpp:533]: Did you test the code? 😅 This breaks the function.

File: src/entities/notefolder.cpp [hunk 2]
```diff
@@ -530,12 +530,12 @@ bool NoteFolder::isPathNoteFolder(const QString &path) {
...
+    settings.setValue(QStringLiteral("NoteFolder-%1/%2").arg(id).arg(key), value);
 }
 
 QVariant NoteFolder::settingsValue(const QString &key, const QVariant &defaultValue) const {
     const SettingsService settings;
-    return settings.value(QString("NoteFolder-%1/%2").arg(QString::number(id), key), defaultValue);
```
Comment (pbek) [inline src/entities/notefolder.cpp:538]: Same here.

File: src/entities/notefolder.cpp (hunk 1 above)
Comment (Integral-Tech) [reply src/entities/notefolder.cpp:533]: Oh, sorry :(

Comment (pbek) [reply src/entities/notefolder.cpp:533]: At least it started to throw warnings and I have a bad feeling, can you please stick to just replacing the QString. 😉

Comment (Integral-Tech) [reply src/entities/notefolder.cpp:533]: So... are failed tests due to this change?

Comment (pbek) [reply src/entities/notefolder.cpp:533]: There are no tests for testing if your change breaks the string. But it seems ok. Thank you.

**Pull request merged**